File: frontend/src/types/Destination.ts
import { SnorkelSite } from "./SnorkelSite";
import { StormRecord } from "./StormRecord ";
import { WeatherData } from "./WeatherData ";
import { FishLifeScore } from "./FishLifeScore ";
import { FieldReport } from "./FieldReport ";
import { DestinationImage } from "./DestinationImage";
import { BestTimeScore } from "./BestTimeScore";

export interface Destination {
  id: number;
  name: string;
  country: string;
  description: string;

  primaryImage: string | null;
  heroImages: string[];
  gallery: string[];
  images: DestinationImage[];

  species: string[];
  videos: string[];
  guide: string;

  sites: SnorkelSite[];
  fishLife: FishLifeScore | null;
  stormHistory: StormRecord[];
  weather: WeatherData | null;

  bestTime: BestTimeScore[];

  mapEmbed: string;
  reports: FieldReport[];
}

File: backend/Api/Models/Destination.cs
using Api.Models;

namespace Api.Models
{
    public class Destination
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Country { get; set; }
        public string Description { get; set; }

        public List<string> Species { get; set; } = new();
        public List<string> Videos { get; set; } = new();

        public string Guide { get; set; }
        public string MapEmbed { get; set; }

        public List<DestinationImage> Images { get; set; } = new();
        public List<SnorkelSite> Sites { get; set; } = new();
        public FishLifeScore FishLife { get; set; }
        public List<StormRecord> StormHistory { get; set; } = new();
        public WeatherData Weather { get; set; }
        public List<FieldReport> Reports { get; set; } = new();
        public List<BestTimeScore> BestTime { get; set; } = new();

    }
}